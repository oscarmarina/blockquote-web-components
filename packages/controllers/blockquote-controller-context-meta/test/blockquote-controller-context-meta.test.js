/* eslint-disable import/no-extraneous-dependencies */
import { html, fixture, expect, fixtureCleanup } from '@open-wc/testing';
import './elements.js';
import './flow-element.js';

suite('BlockquoteControllerContextMeta', () => {
  /**
   * @type {import('./elements').ProviderEl}
   */
  let el;

  teardown(() => fixtureCleanup());

  suite('Default', () => {
    setup(async () => {
      el = await fixture(html`
        <provider-el>
          <consumer-el></consumer-el>
        </provider-el>
      `);
    });

    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        await expect(el).shadowDom.to.equalSnapshot();
      });

      test('LIGHT DOM - Structure test', async () => {
        await expect(el).lightDom.to.equalSnapshot();
      });
    });
  });

  suite('Override ', () => {
    setup(async () => {
      el = await fixture(html`
        <provider-el data="specified from attribute">
          <consumer-el></consumer-el>
        </provider-el>
      `);
    });
    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        await expect(el).shadowDom.to.equalSnapshot();
      });

      test('LIGHT DOM - Structure test', async () => {
        await expect(el).lightDom.to.equalSnapshot();
      });
    });
  });

  suite('Extend', () => {
    setup(async () => {
      el = await fixture(html`
        <flow-element></flow-element>
      `);
    });

    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        await expect(el).shadowDom.to.equalSnapshot();
      });

      test('LIGHT DOM - Structure test', async () => {
        await expect(el).lightDom.to.equalSnapshot();
      });
    });
  });
});
