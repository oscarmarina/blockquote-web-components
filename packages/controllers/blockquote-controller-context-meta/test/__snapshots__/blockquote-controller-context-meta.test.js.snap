// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockquoteControllerContextMeta > Default > Semantic Dom > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<provider-el>
  <consumer-el data-callback="Initial">
  </consumer-el>
</provider-el>
"
`;

exports[`BlockquoteControllerContextMeta > Default > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<h3>
  Provider's data:
  <code>
    Initial
  </code>
</h3>
<slot>
</slot>
"
`;

exports[`BlockquoteControllerContextMeta > Override  > Semantic Dom > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<provider-el data="specified from attribute">
  <consumer-el data-callback="specified from attribute">
  </consumer-el>
</provider-el>
"
`;

exports[`BlockquoteControllerContextMeta > Override  > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<h3>
  Provider's data:
  <code>
    specified from attribute
  </code>
</h3>
<slot>
</slot>
"
`;

exports[`BlockquoteControllerContextMeta > initOrGetContextProvider > Semantic Dom > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<flow-element role="presentation">
</flow-element>
"
`;

exports[`BlockquoteControllerContextMeta > initOrGetContextProvider > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<slot>
</slot>
"
`;
